Sources/FoundationEssentials/WinSDK+Extensions.swift

```diff
@@ -288,4 +296,29 @@ internal func WIN32_FROM_HRESULT(_ hr: HRESULT) -> DWORD {
     return DWORD(hr)
 }
 
+internal func FillNullTerminatedWideStringBuffer(initialSize: DWORD, maxSize: DWORD, _ body: (UnsafeMutableBufferPointer<WCHAR>) throws -> DWORD) throws -> String {
+    var bufferCount = max(1, min(initialSize, maxSize))
+    while bufferCount < maxSize {
```

Comment:
Should this be `bufferCount <= maxSize` so that we actually try max size? (I might be getting confused here as to whether this size includes the null byte or not - perhaps some documentation on this function might help clarify that)

Reply:
The buffer includes the null byte. Yes, I guess this should be <=, good catch.

Reply:
Fixed, added comments